tests/integration/testlib/statelib.py

```diff
@@ -0,0 +1,52 @@
+#
+# Copyright 2018 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+from libnmstate import netinfo
+
+
+INTERFACES = 'interfaces'
+
+
+def show_only(ifnames):
+    """
+    Report the current state, filtering based on the given interface names.
+    """
+    base_filter_state = {
+        INTERFACES: [{'name': ifname} for ifname in ifnames]
+    }
+    return filtered_current_state(base_filter_state)
+
+
+def filtered_current_state(desired_state):
+    """
+    Given the desired state, return a filtered version of the current state
+    which includes only entities that have been mentioned in the desired state.
+    In case there are no interfaces for filtering, all are reported
```

Comment:
The doctext mentions entities without explaining what they are and used interfaces in the next sentence, maybe use `entities such as interfaces` and then `entities` in the second sentence.

Reply:
Done

Reply:
Also IMHO filtered is to unpecific for the function. Since it works on entities, this should be part of the name.

Reply:
Could you please explain what you mean? It should probably be a verb: `filter_current_state`. And it works on a state, I think it is obvious the state has entities.